devtools/onion.c

```diff
@@ -25,23 +26,56 @@ static void do_generate(int argc, char **argv,
 	memset(&sessionkey, 'A', sizeof(sessionkey));
 
 	for (int i = 0; i < num_hops; i++) {
-		if (!hex_decode(argv[1 + i], 66, privkeys[i], 33)) {
+		size_t klen = strcspn(argv[1 + i], "/");
+		if (!hex_decode(argv[1 + i], klen, privkeys[i], 32))
```

Comment:
why did this go from `33` to `32`?

Reply:
Because it was completely broken/untested :(

A *privkey* must be 32 bytes.  A *pubkey* is 33 bytes.

Reply:
I added a Note to the commit msg.